src/cli.ts

```diff
@@ -59,10 +62,93 @@ const noop = (): void => {
+                        as: 'questions',
+                        include: [
+                            {
+                                model: CourseQuestionAssessmentInfo,
+                                as: 'courseQuestionAssessmentInfo'
+                            }
+                        ]
+                    }
+                ]
+            }
+        );
+
+        logger.info(`Checking ${topics.length} topics`);
+
+        await topics.asyncForEach(async (topic) => {
+            // For each question in the topic, sum the existence of a path problem.
+            let count = 0;
+            await topic.questions?.asyncForEach(async (question)=>{
+                const {webworkQuestionPath, errors, courseQuestionAssessmentInfo} = question;
+                const additionalProblemPaths = courseQuestionAssessmentInfo?.additionalProblemPaths;
+                const additionalErrors = courseQuestionAssessmentInfo?.errors;
+
+                const isDefinitelyBad = VALID_PG_PATH_REGEX.test(webworkQuestionPath) === false;
+
+                // If this is definitely bad and not already in the error object, add it.
+                if (isDefinitelyBad && (_.isNil(errors) || _.isEmpty(errors) || !(webworkQuestionPath in errors))){
+                    logger.debug(`Updating ${question.id} from topic ${topic.id} with a path error.`);
+                    question.errors = {
+                        [webworkQuestionPath]: [`${webworkQuestionPath} cannot be found.`]
+                    };
+                    question.save();
+                }
+
+                if (topic.topicTypeId === 2) {
+                    if (_.isNil(courseQuestionAssessmentInfo)) {
+                        logger.error(`Topic ${topic.id} is an exam but this question has no assessment info.`);
+                        if (isDefinitelyBad) count += 1;
+                        return;
+                    }
+                    let atLeastOneIsBad = false;
+                    await additionalProblemPaths?.asyncForEach(async (path) => {
+                        const isDefinitelyBad = VALID_PG_PATH_REGEX.test(path) === false;
+                        if (isDefinitelyBad) atLeastOneIsBad = true;
+                        
+                        if (isDefinitelyBad && (_.isNil(additionalErrors) || _.isEmpty(additionalErrors) || !(path in additionalErrors))) {
+                            if (_.isNil(courseQuestionAssessmentInfo.errors)) {
+                                courseQuestionAssessmentInfo.errors = {};
+                            }
+                            logger.debug(`Updating ${courseQuestionAssessmentInfo.id} from topic ${topic.id} with an additional path error.`);
+                            courseQuestionAssessmentInfo.errors[path] = [`${path} cannot be found`];
+                        }
+                    });
+                    courseQuestionAssessmentInfo.save();
+                    if (isDefinitelyBad || atLeastOneIsBad) count += 1;
```

Comment:
I thought we were counting the number of questions that had errors, not the number of errors on the question

If I had one question with 3 paths for an exam and they were all bad, shouldn't that only be one error on the topic?

This looks like it would add 3 which would be the number of bad paths, I'm fine either way as long as we are consistent

Reply:
This should be outside of the `asyncForEach`, so it should only increment the topic.errors **once** if either the question path or one of the assessment paths are bad.